### ISSUE TITLE
Decouple from refinerycms-authentication-devise

### ISSUE THREAD

**Issue by bricesanchez** [refinery/refinerycms-blog#446]
At this time we have to use Blog at least with refinerycms authentication devise gem : https://github.com/refinery/refinerycms-blog/blob/master/lib/refinery/blog/configuration.rb#L28-L35

We can't use it without an authentication system (like for development or testing).

Perhaps we should be able to disable by default the author feature?

**Comment by parndt** [comment]
@bricesanchez can't we access `authorisation_manager.current_user` or whatever that API was?

**Comment by parndt** [comment]
@bricesanchez I think we should remove the author feature and implement it as purely a string field

**Comment by bricesanchez** [comment]
I'm ok, we should rethink the author feature. I don't think it should be only a string, i use this feature with refinerycms-teams (Refinery::Team model) instead of Refinery::User. It's really useful.

**Comment by parndt** [comment]
So do we disable it unless something is specified? Opt-in to feature?

**Comment by bricesanchez** [comment]
@parndt i think yes.

**Comment by parndt** [comment]
I had to make a fix so that it would boot in current applications.
The problem is, it makes the build fail because we don't depend on refinerycms-authentication-devise and nor should we

**Comment by bricesanchez** [comment]
Could we depend on refinerycms-authentication-devise only to run tests ?
And another solution in the next stable version ?

**Comment by parndt** [comment]
@bricesanchez I don't think so.. but I have a branch where I attempt to solve this entirely: https://github.com/refinery/refinerycms-blog/compare/refinery-user-coupling

**Comment by parndt** [comment]
See #449

**Comment by marcelotto** [comment]
What's the status on this issue? On trying to integrate Refinery with Refinery-Blog in a Spree-app using Spree's authentication system via [spree-refinery-authentication](https://github.com/refinerycms-contrib/spree-refinery-authentication) instead of refinerycms-authentication-devise I was running in issue <https://github.com/refinery/refinerycms-blog/issues/452>. Are there any preliminary workarounds I could try to get this setup running?

**Comment by bricesanchez** [comment]
@marcelotto At this time i manually create the blog initializer before run the extension generator : https://github.com/bricesanchez/spree-refinerycms-test/commit/1e13ebd34f129c6ed4240bae1ac12c806c561ae4

**Comment by marcelotto** [comment]
@bricesanchez Thanks, that helped. 👍

**Comment by bricesanchez** [comment]
@parndt Should we change this line : 
https://github.com/refinery/refinerycms-blog/blob/master/lib/refinery/blog/configuration.rb#L29

for : 

```ruby
class_name = @@user_class_name || "Refinery::Core::NilUser"
```

**Comment by bricesanchez** [comment]
@marcelotto nice to know that there are spree-refinery-authentication users :)

**Comment by marcelotto** [comment]
A bit too early happy. When trying to create a post, it failed on the `collection_select` in `/app/views/refinery/blog/admin/posts/_form.html.erb`, since `Spree::User` has no `username`. Could workaround this by extending `Spree::User` with an appropriate alias:

```ruby
Spree::User.class_eval do
  alias_attribute :username, :login
end
```

In a `refinerycms-authentication-devise`-decoupled solution it would be instrumental to have a config option to set the username attribute.

@bricesanchez `spree-refinery-authentication` works like a charm, thanks.

**Comment by bricesanchez** [comment]
Extending `Spree::User` is the good way.

 `refinerycms-blog` could be have a config to set the username attribute.

Feel free to open a pull request :)

**Comment by parndt** [comment]
> @parndt Should we change this line : 
> https://github.com/refinery/refinerycms-blog/blob/master/lib/refinery/blog/configuration.rb#L29
> 
>  - class_name = @@user_class_name || "Refinery::Authentication::Devise::User"
>  + class_name = @@user_class_name || "Refinery::Core::NilUser"

@bricesanchez Probably? Does it work?

**Comment by bricesanchez** [comment]
Closed in favor of #459
